### PR TITLE
CSS3 vars theming

### PR DESCRIPTION
**Pull request by meodai**
Ohaio cosamias and arigato cosaimas for this awesome project!
 
Referring to #8:

```css
body {
   --tp-font-family: Arial; //overwrite the font family
   --tp-fg-color: pink;
   --tp-input-bg-color: black;
   --tp-input-border-radius: 10px;
    // etc...
}
```

for now I just translated the SCSS vars to CSS vars with fallbacks. I think I can further simplify to make theming super easy for the end user. (Light / Dark / Transparent theme etc..)

PS: those can also be set dynamically. Useful when you want tweakpane to have the best possible contrast to your current visual.

```js
document.documentElement.style.setProperty('--tp-input-bg-color', 'black');
```

### CODE REVIEW

**Comment by cocopon** [comment]
Thank you for the pull request. The basic idea looks good:)

- Please check `npm test` is passing. Maybe needs `npm run main:format` (I'll update CI settings later...)
- Your pull request contains multiple topics (e.g. added `npm start`, `will-change`,  etc.) but I recommend a single commit with a single topic. It makes isolating problems easily. 
- Color variables looks not intuitive and bit difficult to understand... (e.g. `color-light-hs` and `color-light-l`. It's not your fault, it's CSS issue)

File: src/main/sass/_defs.scss [hunk 1]
```diff
@@ -123,27 +267,27 @@ $slider-input-bar-color: rgba($fg-color, 0.2);
...
+		var(--fg-color) 2px,
 		transparent 2px,
 		transparent 4px,
-		$fg-color 4px,
-		$fg-color
+		var(--fg-color) 4px,
+		var(--fg-color)
 	);
-	border-radius: 2px;
+	border-radius: var(--input-border-radius);
```
Comment (cocopon) [inline src/main/sass/_defs.scss]: The border radius of folder marks is not equal to `input-border-radius` (it happened to be `2px`).

**Comment by meodai** [comment]
thanks for the feedback. The variable names are the ones you used in the SCSS also not my favourite names...

**Comment by cocopon** [comment]
I think many (and complex) options causes user confusion. How about reducing the number of them?

Options that I think appropriate:

- Font family (and size?)
- Base hue

**Comment by meodai** [comment]
I am already working on a proposal. I also think some simplification would
be great

**Comment by cocopon** [comment]
Based on this discussion, I implemented the feature and released in 1.3.0. Thank you for the great idea!
https://cocopon.github.io/tweakpane/misc.html#theming

**Pull request closed without merging**